exec/vector/src/main/java/org/apache/drill/exec/vector/complex/MapVector.java

```diff
@@ -317,6 +317,12 @@ public Object getObject(int index) {
         if (v != null && index < v.getAccessor().getValueCount()) {
           Object value = v.getAccessor().getObject(index);
           if (value != null) {
+            if ((v.getAccessor().getObject(index) instanceof Map
+                    && ((Map) v.getAccessor().getObject(index)).size() == 0)
+                || (v.getAccessor().getObject(index) instanceof List
+                    && ((List) v.getAccessor().getObject(index)).size() == 0)) {
+              continue;
+            }
```

Comment:
See the JIRA entry for more comments. The key problem is that Drill does not support standard JSON rules. So, we are simply moving the problem around.

Consider this case:

{code}
{ }
{ "a": { } }
{ "a": { "b": {} } }
{code}

It appears that the code here would emit:
{ }
{ }
{ "a": { } }
{code}

If we are going to apply the empty-is-not-present rule we should do so recursively.